lib/fog/ovirt/requests/compute/v4/create_vm.rb

```diff
@@ -9,18 +9,48 @@ def check_for_option(opts, name)
 
           # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
           def process_vm_opts(opts)
-            return unless check_for_option(opts, "template") && check_for_option(opts, "storagedomain")
+            return unless check_for_option(opts, "template") && check_for_option(opts, :disks)
 
-            template_id = opts[:template] || client.system_service.templates_service.search(:name => opts[:template_name]).first.id
-            template_disks = client.system_service.templates_service.template_service(template_id).get.disk_attachments
-            storagedomain_id = opts[:storagedomain] || storagedomains.select { |s| s.name == opts[:storagedomain_name] }.first.id
+            template_id = opts[:template].id || client.system_service.templates_service.search(:name => opts[:template_name]).first.id
+            template_disks = client.system_service.templates_service.template_service(template_id).disk_attachments_service.list
 
             # Make sure the 'clone' option is set if any of the disks defined by
             # the template is stored on a different storage domain than requested
             opts[:clone] = true unless opts[:clone] == true || template_disks.empty? || template_disks.all? { |d| d.storage_domain == storagedomain_id }
 
-            # Create disks map
-            opts[:disks] = template_disks.collect { |d| { :id => d.id, :storagedomain => storagedomain_id } }
+
+            # Create disks map "clone" from disks, rather then template
+            if opts[:disks] && !opts[:disks].empty?
+              disk_attachments = []
+              opts[:disks].each do |disk|
+                active = disk["active"].present? ? disk["active"] == "true" : true
+                sparse = disk["sparse"].present? ? disk["sparse"] == "true" : true
+                interface = disk["interface"] || OvirtSDK4::DiskInterface::VIRTIO
```

Comment:
@iNecas we now know that the sdk can't change disk interface and provisioned size in 1 api call with the cloning because the ovirt api does not expect this.
Do you think I should delete them for now or comment out with a comment?
There is no RFE to add this to the api for now but a comment with the RFE link if one is opened cold be helpful.

Reply:
+1 for creating and RFE and linking from here